src/sentry/api/event_search.py

```diff
@@ -1431,6 +1442,78 @@ def has_default(self, params):
         "transform": u"abs(minus({column}, {target:g}))",
         "result_type": "duration",
     },
+    # These range functions for performance trends, these aren't If functions
+    # to avoid allowing arbitrary if statements
+    # Not yet supported in Discover, and shouldn't be added to fields.tsx
+    "percentile_range": {
+        "name": "percentile_range",
+        "args": [
+            DurationColumn("column"),
+            NumberRange("percentile", 0, 1),
+            DateArg("start"),
+            DateArg("end"),
+            NumberRange("index", 1, None),
+        ],
+        "aggregate": [
+            u"quantileIf({percentile:.2f})({column},and(greaterOrEquals(timestamp,toDateTime('{start}')),less(timestamp,toDateTime('{end}'))))",
+            None,
+            "percentile_range_{index:g}",
+        ],
+        "result_type": "duration",
+    },
+    "avg_range": {
+        "name": "avg_range",
+        "args": [
+            DurationColumn("column"),
+            DateArg("start"),
+            DateArg("end"),
+            NumberRange("index", 1, None),
+        ],
+        "aggregate": [
+            u"avgIf({column},and(greaterOrEquals(timestamp,toDateTime('{start}')),less(timestamp,toDateTime('{end}'))))",
+            None,
+            "avg_range_{index:g}",
+        ],
+        "result_type": "duration",
+    },
+    "user_misery_range": {
+        "name": "user_misery_range",
+        "args": [
+            NumberRange("satisfaction", 0, None),
+            DateArg("start"),
+            DateArg("end"),
+            NumberRange("index", 1, None),
+        ],
+        "calculated_args": [{"name": "tolerated", "fn": lambda args: args["satisfaction"] * 4.0}],
+        "aggregate": [
+            u"uniqIf(user,and(greater(duration,{tolerated:g}),and(greaterOrEquals(timestamp,toDateTime('{start}')),less(timestamp,toDateTime('{end}')))))",
```

Comment:
Interesting one here, this had to be
```
and(duration>tolerated,
    and(
         timestamp > start,
         timestamp < end
     )
)
```
Because it looks like `and` only supports two terms.

Reply:
Clickhouse supports any number of and terms, but snuba expects everything to be binary. I'm not sure 100% why this wouldn't work for you, since this should be getting passed straight to clickhouse.